src/Microsoft.ML.Transforms/FourierDistributionSampler.cs

```diff
@@ -5,205 +5,252 @@
 using System;
 using Microsoft.ML;
 using Microsoft.ML.CommandLine;
-using Microsoft.ML.EntryPoints;
+using Microsoft.ML.Data;
 using Microsoft.ML.Internal.Utilities;
 using Microsoft.ML.Model;
+using Microsoft.ML.Numeric;
 using Microsoft.ML.Transforms;
 
-[assembly: LoadableClass(typeof(GaussianFourierSampler), typeof(GaussianFourierSampler.Options), typeof(SignatureFourierDistributionSampler),
-    "Gaussian Kernel", GaussianFourierSampler.LoadName, "Gaussian")]
+[assembly: LoadableClass(typeof(GaussianRngGenerator), typeof(GaussianRngGenerator.Options), typeof(SignatureRngGenerator),
+    "Gaussian Kernel", GaussianRngGenerator.LoadName, "Gaussian")]
 
-[assembly: LoadableClass(typeof(LaplacianFourierSampler), typeof(LaplacianFourierSampler.Options), typeof(SignatureFourierDistributionSampler),
-    "Laplacian Kernel", LaplacianFourierSampler.RegistrationName, "Laplacian")]
+[assembly: LoadableClass(typeof(LaplacianRngGenerator), typeof(LaplacianRngGenerator.Options), typeof(SignatureRngGenerator),
+    "Laplacian Kernel", LaplacianRngGenerator.LoadName, "Laplacian")]
 
 // This is for deserialization from a binary model file.
-[assembly: LoadableClass(typeof(GaussianFourierSampler), null, typeof(SignatureLoadModel),
-    "Gaussian Fourier Sampler Executor", "GaussianSamplerExecutor", GaussianFourierSampler.LoaderSignature)]
+[assembly: LoadableClass(typeof(GaussianRngGenerator.RandomNumberGenerator), null, typeof(SignatureLoadModel),
+    "Gaussian Fourier Sampler Executor", "GaussianSamplerExecutor", GaussianRngGenerator.RandomNumberGenerator.LoaderSignature)]
 
 // This is for deserialization from a binary model file.
-[assembly: LoadableClass(typeof(LaplacianFourierSampler), null, typeof(SignatureLoadModel),
-    "Laplacian Fourier Sampler Executor", "LaplacianSamplerExecutor", LaplacianFourierSampler.LoaderSignature)]
+[assembly: LoadableClass(typeof(LaplacianRngGenerator.RandomNumberGenerator), null, typeof(SignatureLoadModel),
+    "Laplacian Fourier Sampler Executor", "LaplacianSamplerExecutor", LaplacianRngGenerator.RandomNumberGenerator.LoaderSignature)]
 
 // REVIEW: Roll all of this in with the RffTransform.
 namespace Microsoft.ML.Transforms
 {
     /// <summary>
-    /// Signature for an IFourierDistributionSampler constructor.
+    /// Signature for a <see cref="RngGeneratorBase"/> constructor.
     /// </summary>
     [BestFriend]
-    internal delegate void SignatureFourierDistributionSampler(float avgDist);
+    internal delegate void SignatureRngGenerator();
 
-    public interface IFourierDistributionSampler : ICanSaveModel
+    public abstract class RngGeneratorBase
     {
-        float Next(Random rand);
+        internal abstract float Dist(in VBuffer<float> first, in VBuffer<float> second);
+
+        internal abstract RandomNumberGeneratorBase GetRandomNumberGenerator(float avgDist);
```

Comment:
What's the difference between `RngGeneratorBase` and `RandomNumberGeneratorBase`?